File: 1107-minimum-swaps-to-group-all-1s-together/minimum-swaps-to-group-all-1s-together.cpp
class Solution {
public:
    int minSwaps(vector<int>& data) {

        int ones = count(data.begin(),data.end(),1);
        int zeros = 0;
        int n = data.size();

        int r = 0;
        for (; r != ones; ++r) if (data[r] == 0) ++zeros;
        int ans = zeros;
        for (; r != n;++r) {
            if (data[r] == 0) ++zeros;
            if (data[r-ones] == 0) --zeros;
            ans = min(ans, zeros);

        }

        return ans;
        
    }
};

File: 1107-minimum-swaps-to-group-all-1s-together/minimum-swaps-to-group-all-1s-together.py
class Solution:
    def minSwaps(self, data: List[int]) -> int:
        

        ones = data.count(1)

        zeros = 0
        l = r = 0
        n = len(data)
        for r in range(ones):
            if data[r] == 0:
                zeros += 1
        ans = zeros
        
        while r < n-1:
            r += 1
            if data[r] == 0:
                zeros += 1
            if data[l] == 0:
                zeros -= 1
            l += 1
            ans= min(ans,zeros)
        return ans